src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -83,15 +56,30 @@
   If the query can't be parsed, a 400 is returned."
   [entity version query paging-options db url-prefix]
   (try
-    (let [parsed-query (json/parse-strict-string query true)]
-      (stream-query-result entity version parsed-query paging-options db url-prefix http/stream-json-response))
+    (let [[query->sql munge-fn] (entity->sql-fns entity version paging-options url-prefix)
+          {:keys [results-query count-query]} (jdbc/with-transacted-connection db
```

Comment:
Previously we would use one connection for this query and the one on line 75. Is there a reason to use two now?